api/src/test/java/org/openmrs/api/db/ContextDAOTest.java

```diff
@@ -309,7 +311,13 @@ public void authenticate_shouldPassRegressionTestFor1580() {
 		
 		// those were the first eight, now the ninth request 
 		// (with the same user and right pw) should fail
-		assertThrows(ContextAuthenticationException.class, () -> dao.authenticate("admin", "test"));
+		assertThrows(ContextAuthenticationException.class, () -> {
+			Context.addProxyPrivilege(PrivilegeConstants.GET_GLOBAL_PROPERTIES);
+			dao.authenticate("admin", "test");
+			Context.removeProxyPrivilege(PrivilegeConstants.GET_GLOBAL_PROPERTIES);
```

Comment:
After removing the proxy privilege above, do we again need to remove it below?

Reply:
Yes, we need to remove it again. The code inside the lambda function only adds or removes the value to that context. I tested this using `Context.hasPrivilege(PrivilegeConstants.GET_GLOBAL_PROPERTIES)`